com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/DirectLinkingResourceStorageLoadable.java

```diff
@@ -328,6 +329,30 @@ protected void loadFeatureValue(final InternalEObject internalEObject, final ESt
           throw e;
         }
       }
+
+      @Override
+      public void loadResource(final Resource resource) throws IOException {
+        this.resource = resource;
+        resourceSet = resource.getResourceSet();
+        URI uri = resource.getURI();
+        if (uri != null && uri.isHierarchical() && !uri.isRelative()) {
+          baseURI = uri;
+        }
+        int size = readCompressedInt();
+        if (!ResourceSetOptions.installDerivedState(resourceSet) && size == 2) { // the InfererenceContainer is always in the second slot

```

Comment:
I take it this means we do not believe there may be cases with size > 2?

Reply:
That is correct